stdlib/buffer.mli

```diff
@@ -131,3 +131,23 @@ val add_channel : t -> in_channel -> int -> unit
 val output_buffer : out_channel -> t -> unit
 (** [output_buffer oc b] writes the current contents of buffer [b]
    on the output channel [oc]. *)
+
+
+(** {6 Iterators} *)
+
+val to_iter : t -> char Iter.t
+(** Iterate on the buffer, in increasing order
```

Comment:
The semantics w.r.t. `Buffer.clear` and `Buffer.reset` should be specified (or explicitly left undefined).